File: car.py
class Car:
    def __init__(self, make, model, color, price):
        self.make = make
        self.model = model
        self.color = color
        self.price = price
        self.mileage = 0

    # instance method
    def set_price(self, price):
        self.price += price
        print(f'the price of the car is now {self.price}')
    
    def paint(self, color):
        self.color = color
        print(f'painting the car {self.color}!')
    
    def show_car_info(self):
        print(f'make: {self.make}\nmodel: {self.make}\ncolor: {self.color}\nprice: ${self.price}\nmileage: {self.mileage}')

    def travel(self, distance):
        self.mileage += distance
        print(f'you traveled {distance}, your mileage is now at {self.mileage} miles!')

File: lib/car.rb
class Car
  attr_accessor :make, :model, :color, :price, :mileage

  def initialize(make, model, color, price)
    @make = make
    @model = model
    @color = color
    @price = price
    @mileage = 0
  end

  def set_price(price)
    @price += price
    puts "the price of the car is now #{@price}"
  end

  def paint(color)
    @color = color
    puts "painting the car #{@color}"
  end

  def show_car_info
    puts "make: #{@make}\nmodel: #{@model}\ncolor: #{@color}\nprice: #{@price}\nmileage: #{@mileage}"
  end

  def travel(distance)
    @mileage += distance
    puts "you traveled #{@mileage}, your mileage is now at #{@mileage} miles!"
  end
end
